src/settings.rs

```diff
@@ -78,48 +62,28 @@ mod tests {
 
     #[test]
     fn validate_empty_rules_settings() -> Result<(), ()> {
-        let settings = Settings { rules: vec![] };
-        assert!(
-            settings.validate().is_err(),
-            "Settings should contains at least one rule"
-        );
-        Ok(())
-    }
-    #[test]
-    fn validate_rule_with_no_envvar_settings() -> Result<(), ()> {
-        let settings = Settings {
-            rules: vec![Rule {
-                environment_variables: HashSet::from([]),
-                ..Default::default()
-            }],
+        let settings = Settings::ContainsAllOf {
+            envvars: HashSet::new(),
         };
         assert!(
             settings.validate().is_err(),
-            "Settings should not accept rule with no environment variables"
+            "Empty environment variable list is not allowed"
         );
         Ok(())
     }
 
     #[test]
     fn settings_should_allow_envvar_name() -> Result<(), ()> {
```

Comment:
This test does not need to return anything, we can get rid of the `Result<(), ()>`.

Also, can we merge the two tests into a single table test?